app/scripts/migrations/133.2.test.ts

```diff
@@ -0,0 +1,185 @@
+  });
+
+  it('Removes tokens with empty address', async () => {
+    const oldStorage = {
+      meta: { version: oldVersion },
+      data: {
+        TokensController: {
+          allTokens: {
+            '0x1': {
+              '0x123': [
+                {
+                  address: '0x0000000000000000000000000000000000000000',
+                  symbol: 'eth',
+                  decimals: 18,
+                },
+                { address: '0x2', symbol: 'TOKEN2', decimals: 18 },
+              ],
+            },
+          },
+        },
+      },
+    };
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.data).toStrictEqual({
+      TokensController: {
+        allTokens: {
+          '0x1': {
+            '0x123': [{ address: '0x2', symbol: 'TOKEN2', decimals: 18 }],
+          },
+        },
+      },
+    });
+  });
+
+  it('Removes tokens with empty address across multiple accounts', async () => {
+    const oldStorage = {
+      meta: { version: oldVersion },
+      data: {
+        TokensController: {
+          allTokens: {
+            '0x1': {
+              '0x123': [
+                {
+                  address: '0x0000000000000000000000000000000000000000',
+                  symbol: 'eth',
+                  decimals: 18,
+                },
+                { address: '0x2', symbol: 'TOKEN2', decimals: 18 },
+              ],
+              '0x456': [
+                {
+                  address: '0x0000000000000000000000000000000000000000',
+                  symbol: 'eth',
+                  decimals: 18,
+                },
+                { address: '0x3', symbol: 'TOKEN3', decimals: 18 },
+              ],
+              '0x789': [{ address: '0x4', symbol: 'TOKEN4', decimals: 18 }],
+            },
+          },
+        },
+      },
+    };
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.data).toStrictEqual({
+      TokensController: {
+        allTokens: {
+          '0x1': {
+            '0x123': [{ address: '0x2', symbol: 'TOKEN2', decimals: 18 }],
+            '0x456': [{ address: '0x3', symbol: 'TOKEN3', decimals: 18 }],
+            '0x789': [{ address: '0x4', symbol: 'TOKEN4', decimals: 18 }],
+          },
+        },
+      },
+    });
+  });
+
+  it('Does not change state on chains other than mainnet', async () => {
+    const oldStorage = {
+      meta: { version: oldVersion },
+      data: {
+        TokensController: {
+          allTokens: {
+            '0x999': {
+              '0x123': [
+                {
+                  address: '0x0000000000000000000000000000000000000000',
+                  symbol: 'eth',
+                  decimals: 18,
+                },
+                { address: '0x2', symbol: 'TOKEN2', decimals: 18 },
+              ],
+            },
+          },
+        },
+      },
+    };
+    const newStorage = await migrate(oldStorage);
+    expect(newStorage.data).toStrictEqual(oldStorage.data);
+  });
```

Comment:
Still think that we may need to do this on other chains, but I guess we can cross that bridge if we come to it?

I guess I don't understand why this bug would only be apparent on mainnet, when the zero balance issue was showing up across different networks as well.

Reply:
Just saw your comment further down. 👍

Reply:
Polygon for example has their native token as an erc-20-esque contract deployed at `0x0000000000000000000000000000000000001010`.